optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/shift/Shift.java

```diff
@@ -99,6 +106,52 @@ public boolean precedes(Shift other) {
         return !endDateTime.isAfter(other.startDateTime);
     }
 
+    public long getLengthInHours() {
+        if (lengthInHours == null) {
+            lengthInHours = startDateTime.until(endDateTime, ChronoUnit.HOURS);
+        }
+        return lengthInHours;
+    }
+
+    public static long calculateLoad(Collection<Integer> hourlyCounts) {
+        long sumSquares = 0;
+        for (int hourlyCount: hourlyCounts) {
+            sumSquares += hourlyCount * hourlyCount;
+        }
+        long squareRootOfSums = Math.round(Math.sqrt(sumSquares) * 1000);
```

Comment:
Should the 1000 be here, or in the score constraint file?

Reply:
This is not so much a constraint weight as a way to avoid losing precision. In this context, 2.123 is something far different from 2.0.

Once I convert this code to CS-D, we may be able to make this a `BIgDecimal` easily, and , keep it in its original decimal form and only use constraint configuration to weight it properly.